File: examples/example_usage.js
#!/usr/bin/env node
/**
 * Example script demonstrating how to use the Claude configuration
 */

const fs = require("fs");
const path = require("path");

// Get the base directory
const BASE_DIR = path.join(__dirname, "..");

/**
 * Load the main Claude configuration
 */
function loadConfig() {
	const configPath = path.join(BASE_DIR, ".claude", "config.json");
	const configData = fs.readFileSync(configPath, "utf8");
	return JSON.parse(configData);
}

/**
 * Load a system prompt by name
 */
function loadPrompt(promptName) {
	const promptPath = path.join(
		BASE_DIR,
		".claude",
		"prompts",
		`${promptName}.md`,
	);
	return fs.readFileSync(promptPath, "utf8");
}

/**
 * Load API settings
 */
function loadApiSettings() {
	const settingsPath = path.join(
		BASE_DIR,
		".claude",
		"settings",
		"api_settings.json",
	);
	const settingsData = fs.readFileSync(settingsPath, "utf8");
	return JSON.parse(settingsData);
}

async function main() {
	// Load configuration
	const config = loadConfig();
	console.log(`Model: ${config.model}`);
	console.log(`Max Tokens: ${config.max_tokens}`);
	console.log(`Temperature: ${config.temperature}`);

	// Load a prompt
	const codingPrompt = loadPrompt("coding_assistant");
	console.log(
		`\nLoaded prompt (first 100 chars):\n${codingPrompt.substring(0, 100)}...`,
	);

	// Load API settings
	const apiSettings = loadApiSettings();
	console.log(`\nAPI Base URL: ${apiSettings.base_url}`);
	console.log(`Timeout: ${apiSettings.timeout}s`);

	// Example: Making a request (pseudo-code)
	// This would require: npm install @anthropic-ai/sdk
	/*
    const Anthropic = require('@anthropic-ai/sdk');
    
    const anthropic = new Anthropic({
        apiKey: process.env.ANTHROPIC_API_KEY,
    });
    
    const message = await anthropic.messages.create({
        model: config.model,
        max_tokens: config.max_tokens,
        temperature: config.temperature,
        system: codingPrompt,
        messages: [
            {
                role: 'user',
                content: 'Write a JavaScript function to calculate fibonacci numbers'
            }
        ]
    });
    
    console.log(message.content);
    */
}

if (require.main === module) {
	main().catch(console.error);
}

module.exports = { loadConfig, loadPrompt, loadApiSettings };

File: examples/example_usage.py
#!/usr/bin/env python3
"""
Example script demonstrating how to use the Claude configuration
"""

import json
from pathlib import Path

# Get the base directory
BASE_DIR = Path(__file__).parent.parent


def load_config():
    """Load the main Claude configuration"""
    config_path = BASE_DIR / ".claude" / "config.json"
    with open(config_path, "r") as f:
        return json.load(f)


def load_prompt(prompt_name):
    """Load a system prompt by name"""
    prompt_path = BASE_DIR / ".claude" / "prompts" / f"{prompt_name}.md"
    with open(prompt_path, "r") as f:
        return f.read()


def load_api_settings():
    """Load API settings"""
    settings_path = BASE_DIR / ".claude" / "settings" / "api_settings.json"
    with open(settings_path, "r") as f:
        return json.load(f)


def main():
    # Load configuration
    config = load_config()
    print(f"Model: {config['model']}")
    print(f"Max Tokens: {config['max_tokens']}")
    print(f"Temperature: {config['temperature']}")

    # Load a prompt
    coding_prompt = load_prompt("coding_assistant")
    print(f"\nLoaded prompt (first 100 chars):\n{coding_prompt[:100]}...")

    # Load API settings
    api_settings = load_api_settings()
    print(f"\nAPI Base URL: {api_settings['base_url']}")
    print(f"Timeout: {api_settings['timeout']}s")

    # Example: Making a request (pseudo-code)
    # This would require the anthropic library: pip install anthropic
    """
    import anthropic
    
    client = anthropic.Anthropic(
        api_key=os.environ.get("ANTHROPIC_API_KEY"),
    )
    
    message = client.messages.create(
        model=config['model'],
        max_tokens=config['max_tokens'],
        temperature=config['temperature'],
        system=coding_prompt,
        messages=[
            {"role": "user", "content": "Write a Python function to calculate fibonacci numbers"}
        ]
    )
    
    print(message.content)
    """


if __name__ == "__main__":
    main()
